src/index.ts

```diff
@@ -330,8 +330,18 @@ export default class StatsGatherer extends EventEmitter {
 
   private async gatherStats (): Promise<Array<{key: RTCStatsType, value: any}>> {
     try {
-      const stats = await this.peerConnection.getStats(null).then(this.polyFillStats.bind(this));
-      return stats;
+      if (['connecting', 'connected'].includes(this.peerConnection.connectionState)) {
+        const stats = await this.peerConnection.getStats(null).then(this.polyFillStats.bind(this));
+        return stats;
+      } else if (this.peerConnection.connectionState === 'closed') {
```

Comment:
If there are any other states to check and be concerned about, please let me know.  Also if `iceConnectionState` should be considered as well

Reply:
I think this is fine.